components/org.wso2.carbon.identity.oauth.uma.resource.endpoint/src/main/java/org/wso2/carbon/identity/oauth/uma/resource/endpoint/impl/ResourceRegistrationApiServiceImpl.java

```diff
@@ -203,63 +175,76 @@ public Response updateResource(String resourceId, ResourceDetailsDTO updatedReso
-
-        HandleErrorResponseConstants handleErrorResponseConstants = new HandleErrorResponseConstants();
-        if (handleErrorResponseConstants.getResponseMap().containsKey(umaException.getErrorCode())) {
-            String statusCode = handleErrorResponseConstants.getResponseMap().get(umaException.getErrorCode())[0];
-            Response.Status status = Response.Status.fromStatusCode(Integer.parseInt(statusCode));
-            String errorCode = handleErrorResponseConstants.getResponseMap().get(umaException.getErrorCode())[1];
-
-            throw buildResourceEndpointException(status, errorCode, umaException.getMessage());
+    private void handleErrorResponse(Throwable throwable, boolean isServerException)
+            throws ResourceEndpointException {
+
+        String code;
+        String errorCode = null;
+        Response.Status status = Response.Status.INTERNAL_SERVER_ERROR;
+        boolean isStatusOnly = true;
+        if (throwable instanceof UMAException) {
+            code = ((UMAException) throwable).getErrorCode();
+        } else {
+            code = ResourceConstants.ErrorMessages.ERROR_CODE_UNEXPECTED.getCode();
+        }
+        if (isServerException) {
+            if (throwable == null) {
+                log.error(status.getReasonPhrase());
+            } else {
+                log.error(status.getReasonPhrase(), throwable);
+            }
+        } else {
+            HandleErrorResponseConstants handleErrorResponseConstants = new HandleErrorResponseConstants();
+            if (handleErrorResponseConstants.getResponseMap().containsKey(code)) {
+                String statusCode = handleErrorResponseConstants.getResponseMap().get(code)[0];
+                status = Response.Status.fromStatusCode(Integer.parseInt(statusCode));
+                errorCode = handleErrorResponseConstants.getResponseMap().get(code)[1];
+                isStatusOnly = false;
+            }
         }
+        throw buildResourceEndpointException(status, errorCode, throwable == null ? "" : throwable.getMessage(),
+                isStatusOnly);
     }
 
-    private static ResourceEndpointException buildResourceEndpointException(Response.Status status,
-                                                                            String code, String description) {
+    public ResourceEndpointException buildResourceEndpointException(Response.Status status, String errorCode,
+                                                                    String description, boolean isStatusOnly) {
 
-        ErrorDTO errorDTO = new ErrorDTO();
-        errorDTO.setCode(code);
-        errorDTO.setDescription(description);
-        return new ResourceEndpointException(status, errorDTO);
+        if (isStatusOnly) {
+            return new ResourceEndpointException(status);
+        } else {
+            ErrorDTO errorDTO = new ErrorDTO();
```

Comment:
call the private method  private ErrorDTO getErrorDTO()  here

Reply:
Done.